core/shared/src/main/scala/zio/Hub.scala

```diff
@@ -160,7 +160,7 @@ object Hub {
         ZIO.fiberIdWith { fiberId =>
           shutdownFlag.set(true)
           ZIO
-            .whenZIO(shutdownHook.succeed(())) {
+            .whenZIO(shutdownHook.succeedUnit) {
```

Comment:
Bit outside the scope of this PR, but since we're at it, you can use `whenZIODiscard` to avoid the `Some` allocation and the FlatMap below from calling `unit`

Reply:
Done. I also used `ZIO.foreachParDiscard`